File: Course work/McEwen_Assignment_1/Assignment_1/Assignment1.java
/*******************************************************************************
*                           Assignment 1 – Python to Java I                    *
*                                                                              *
* PROGRAMMER:       Ben-McEwen                                                 *
* CLASS:            CG120                                                      *
* ASSIGNMENT:       Assignment 1                                               *
* INSTRUCTOR:       Dean Zeller                                                *
* SUBMISSION DATE:  Friday, January 18th 2019                                  *
*                                                                              *
* DESCRIPTION:                                                                 *
*  This program applies basic satistical calulations                           *  
* to a series of positive integers.                                            *
*                                                                              *
* COPYRIGHT:                                                                   *
* This program is (c) 2019 your-name and Dean Zeller. This is original work,   *
* without use of outside sources.                                              *
*******************************************************************************/

import java.util.Scanner;
import java.util.ArrayList;

public class Assignment1
{
    public static void main(String[] args) 
    {
        // Declares variables
        int count;
        int len, horizline, num, sum, max, min;
        String response, title;
        ArrayList<String> data = new ArrayList<String>(); // Create an ArrayList object
       
        // requests report title
        Scanner reader1 = new Scanner(System.in);
        System.out.println();       
        System.out.print("Enter a title for this report \n => ");
        title = reader1.nextLine();
        
        // requests report data
        Scanner reader2 = new Scanner(System.in);
        System.out.println();
        System.out.println("Enter numbers, end with a 0.");
        num = 1;
        sum = 0;
        max = -1000;
        min = 1000;
        count = 1;
        
        //requests user defined series of positive integers
        while (num > 0)
        {
            System.out.printf("  num#%d => ", count);
            response = reader2.nextLine();
            num = Integer.parseInt(response);
            
            //stops loop if 0 is entered (notice there are no braces)
            if (num == 0)
                break;
            
            // iterates for responses greater than 0
            if (num > 0)
            {
               data.add(response);
               sum += num;
               count += 1;
               
               ////determines integer with largest value
               if (max < num)
                   max = num;
               
               
               //determines integer with smallest value
               if (min > num)
                   min = num;
            }
        }
        len = data.size();
        horizline = title.length();
        System.out.println();
        
        //converts arraylist to string and allows character replacement
        String datastring = data.toString()
            .replace(",", "")  // replaces commas with nothing
            .replace("[", "")  // replaces open bracket with nothing
            .replace("]", "")  // replaces close bracket with nothing
            .trim();           //eliminates leading and trailing spaces
        
        
        //displays user defined series of positive integers    
        System.out.println(len + " numbers were entered: " + datastring);
        System.out.println();
        
        //declares variable, typecasts int to double for sum and len, calculates average of data
        Double avg = ((double)sum/(double)len);
        
        //prints report
        System.out.println(title);
        for (int i = 0; i < horizline; i++)
            System.out.print("-");
        
        System.out.printf("\n  N:         %-4d"   , len);
        System.out.printf("\n  Sum:       %-4d"   , sum);
        System.out.printf("\n  Average:   %-4s"   , avg);
        System.out.printf("\n  Smallest:  %-4d"   , min);
        System.out.printf("\n  Largest:   %-4d"   , max);
        System.out.printf("\n  Range:     %-4d"   , (max - min));
        System.out.println();
    }
}

File: Course work/McEwen_Assignment_1/Assignment_1/Assignment1.py
################################################################################
#                       Assignment 1 – Python to Java I                        #
#                                                                              #
# PROGRAMMER:       Ben-Mcewen                                                 #
# CLASS:            CS102                                                      #
# ASSIGNMENT:       Assignment 1                                               #
# INSTRUCTOR:       Dean Zeller                                                #
# SUBMISSION DATE:  Friday, January 18th 2019                                  #
#                                                                              #
# DESCRIPTION:                                                                 #
# This program applies basic satistical calulations                            #
# to a series of positive integers.                                            #
#                                                                              #                                                                              
# COPYRIGHT:                                                                   #
# This program is (c) 2019 Ben-McEwen and Dean Zeller. This is original work,  #
# without use of outside sources.                                              #
################################################################################

#declaration of variables
data = []
num = 1
sum = 0
count = 1

#requests a tilte for the report
title = input("Enter a title for this report \n => ")

#requests a series of positive integers, appends them to an array and calculates
#their sum
print()
print("Enter a series of positive numbers, ending with a 0.")

while num > 0:
    response = int(input("  num#%s => " % count))
    num = response
    if num != 0:
        data.append(num)
        sum += num
        count += 1
        
#Displays the user defined series of positive integers along with a count of the
#number of integers entered        
print()
print(len(data), "numbers were entered: ", end = "")
for d in data:
    print(d, end=" ")    
print() 

#determines the integer with the largest value
largest = data[0]
for large in data:
    if large > largest:
        largest = large

#determines the integer with the smallest value
smallest = data[0]
for small in data:
    if small < smallest:
        smallest = small
        
#calculates average        
average = sum/len(data)

#Displays report summary including count, sum, average, min, max, and range.
print()
print(title)
for x in title:
    print("-", end="") #credit to K. Ritter for teaching me this.
print()
print("  N:        %-4d"    % len(data))
print("  Sum:      %-4d"    % sum)
print("  Average:  %-4.1f"  % average)
print("  Smallest: %-4d"    % smallest)
print("  Largest:  %-4d"    % largest)
print("  Range:    %-4d"    % (largest - smallest))
print()
      
